src/OpenTelemetry/BackwardCompatibilityUtils.cs

```diff
@@ -0,0 +1,68 @@
+// <copyright file="BackwardCompatibilityUtils.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry
+{
+    internal static class BackwardCompatibilityUtils
+    {
+        internal const string UnsetStatusCodeTagValue = "UNSET";
+        internal const string OkStatusCodeTagValue = "OK";
+        internal const string ErrorStatusCodeTagValue = "ERROR";
+
+        internal static void SetActivityStatusUsingTags(Activity activity)
+        {
+            ActivityStatusCode statusCode = ActivityStatusCode.Unset;
+            string statusDescription = null;
+            foreach (var tag in activity.TagObjects)
```

Comment:
`foreach` on activity.TagObjects will allocate an enumerator. If you use the `EnumerateTags` extension, you can do it in an allocation-free way. It is a lot more code to use that API, but this is in the hot path so I think it might be worth doing. I can help with that if you want just LMK.